### PR TITLE
Add standalone torch2whirl Python frontend and ResNet ingestion

### PR DESCRIPTION
**Pull request by shinmingliu**
## Summary

- add the standalone `torch2whirl` executable and `open64_dsc` Python package
  under the Open64 build umbrella
- capture PyTorch/FX graphs and ingest a complete LocalResNet operator set
  through opaque native builder handles
- emit first-class binary Very High Level WHIRL plus deterministic external
  tensor payloads in `model.safetensors`
- support model input, tensor constants, external tensor constants,
  convolution, inference batch normalization, pooling, residual add, flatten,
  linear, matmul, and logits output
- keep Python-facing values opaque and preserve frontend/backend isolation
- use temporary files and atomic replacement so failed frontend runs do not
  leave apparently valid partial artifacts

## Integration Status

PR #64 is merged. This branch was rebased onto merge commit `67bc1440` and now
contains only seven frontend-specific commits on top of `develop`; the obsolete
infrastructure merge commit and duplicated main-agent history are gone.

The Python process is a source frontend only. It writes `model.B` through the
native builder and exits. The Open64 driver and middle end reopen the artifact
through the mapped-image/ELF WHIRL path without embedding Python or depending
on Python-facing WN layout.

## Native Certification

- standalone command contract:
  `torch2whirl model.py --entry forward --sample-input shape:1,3,64,64 --backend native --output model.B`
- full LocalResNet `ir_b2a -st` evidence includes every expected logical
  operator, tensor descriptor, result value, external payload reference,
  symbol, and type
- observed full-model evidence includes 44 `cnn.conv2d`,
  7 `cnn.batch_norm_infer`, and 3 `common.residual_add` records
- the synthetic native ir-tools probe now creates operator-specific compatible
  tensor inputs and treats missing released native operators as failures rather
  than compatibility skips

## End-To-End Driver Evidence

Validated with the merged native infrastructure:

- `openpy -keep model.py` produced nonempty `model.B`,
  `model.safetensors`, and post-lowering `model.t`
- `model.t` contained no executable logical DSL operators after
  `VHO_DSL_Lower_Driver()`
- separate-process `opencc -x whirl -O0 -c model.B` completed and produced
  `model.o`

## Validation After Rebase

- Python unit suite: 103 tests, 41 expected host-native skips
- Linux Docker PyTorch/FX lane: 29 tests passed
- standalone torch2whirl PyTorch driver smoke passed
- clean Linux native build: 103 tests passed with no skips
- native Python `ir_b2a -st` operator smoke passed without capability skips
- driver-native LocalResNet `ir_b2a -st` smoke passed
- Python compileall, `git diff --check`, no-tab scan, and backend-isolation scan
  passed

The first attempted native rerun reused an old configured build directory and
exposed a stale pre-extension `opcode.o`. Certification was repeated in a new
configured directory so every opcode, WN, builder, gatekeeper, image, native
bridge, JsonCpp, and ir-tools object was rebuilt from the rebased source.

### CODE REVIEW

**Review by Yvan-xy — APPROVED** [review]
Looks good to me
